### PR TITLE
Add support for scenarios and distributed queries

### PR DESCRIPTION
**Pull request by gpeb2**
This includes adding DCAT concepts to describe data uploaded into the stack and then facilitate the querying of sub-selections of that data.

### CODE REVIEW

File: Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java [hunk 1]
```diff
@@ -1,68 +1,98 @@
...
+import uk.ac.cam.cares.jps.base.derivation.ValuesPattern;
+
 public class Dataset {
 
+    protected static final Logger LOGGER = LoggerFactory.getLogger(Dataset.class);
+
+    private static final String NAME_DEPRECATION_NOTICE = "Specification of '{}' name is deprecated, in the future the name of the dataset will be used, i.e. '{}' instead of '{}'";
+
+    static final String DEFAULT_NAMESPACE = "https://www.theworldavatar.com/kg/";
     static final String DEFAULT_BASE_IRI = "https://www.theworldavatar.com/kg/";
```
Comment (cmclscourtney) [inline Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java:50]: Should we consider using `https://theworldavatar.io/kg`? Or would this break backwards compatibility

File: Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java [hunk 2]
```diff
@@ -125,7 +167,225 @@ public boolean isSkip() {
...
+        Variable ontopServiceVar = query.var();
+
+        Iri catalogIri;
+        if (!exists) {
+            iri = SparqlConstants.DEFAULT_NAMESPACE + UUID.randomUUID();
+            catalogIri = Rdf.iri(iri);
+
+            insertTriples.add(catalogIri.isA(getRdfType())
+                    .andHas(DCTERMS.TITLE, getName())
+                    .andHas(DCTERMS.DESCRIPTION, getDescription())
```
Comment (cmclscourtney) [inline Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java:217]: Would be great to be able to "@" a txt file or something like with other options

File: Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java [hunk 3]
```diff
@@ -125,7 +167,225 @@ public boolean isSkip() {
...
+                        .getString(dataSubsetNameVar.getQueryString().substring(1));
+
+                nameToDataSubsetMap.get(queriedDatasubsetName).setIri(queriedDatasubset);
+                nameToDataSubsetMap.get(queriedDatasubsetName).setExists(true);
+            }
+        }
+    }
+
+    private static class SparqlConstants {
+        static final String DEFAULT_NAMESPACE = "https://www.theworldavatar.com/kg/";
```
Comment (cmclscourtney) [inline Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java:385]:
```suggestion
        static final String DEFAULT_BASE_IRI = "https://www.theworldavatar.io/kg/";
```

File: Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java [hunk 4]
```diff
@@ -125,7 +167,215 @@ public boolean isSkip() {
...
+        static final Iri BLAZEGRAPH = Rdf.iri(DEFAULT_NAMESPACE + "Blazegraph");
+        static final Iri POSTGIS = Rdf.iri(DEFAULT_NAMESPACE + "PostGIS");
+        static final Iri GEOSERVER = Rdf.iri(DEFAULT_NAMESPACE + "GeoServer");
+        static final Iri ONTOP = Rdf.iri(DEFAULT_NAMESPACE + "Ontop");
```
Comment (cmclscourtney) [inline Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java:379]: Perhaps create new issue for adding data about services with these types

File: Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java (hunk 1 above)
Comment (gpeb2) [reply Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java]: Resolved in commit [a0de9b8](https://github.com/cambridge-cares/TheWorldAvatar/pull/732/commits/a0de9b827bedacedb2070789d4b58acf8f6cef7b).

File: Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/utils/JsonHelper.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+public class JsonHelper {
+
+    private JsonHelper() {
+    }
+
+    @Nonnull
+    public static final ObjectMapper getMapper() {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new Jdk8Module());
```
[the remark this answers is not in the document]
Comment (gpeb2) [reply Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/utils/JsonHelper.java]: Resolved in this commit a8837b9.

File: Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java (hunk 3 above)
Comment (gpeb2) [reply Deploy/stacks/dynamic/stack-clients/src/main/java/com/cmclinnovations/stack/clients/core/datasets/Dataset.java]: Resolved in commit a0de9b8.

**Review by cmclscourtney — APPROVED** [review]
Review and tested as part of CReDo and otherwise

**Review by cmclscourtney — APPROVED** [review]

**Review by cmclscourtney — APPROVED** [review]

**Review by cmclscourtney — APPROVED** [review]
Latest commit looks good

**Pull request merged**